packages/ember-metal/index.d.ts

```diff
@@ -6,7 +6,17 @@ export const PROPERTY_DID_CHANGE: symbol;
 
 export function setHasViews(fn: () => boolean): null;
 
-export function runInTransaction(context: any, methodName: string): any;
+export type MethodKey<T> = { [K in keyof T]: T[K] extends (() => void) ? K : never }[keyof T];
+export type RunInTransactionFunc = <T extends object, K extends MethodKey<T>>(
+  context: T,
+  methodName: K
+) => boolean;
+export type DidRenderFunc = (object: any, key: string, reference: any) => void;
+export type AssertNotRenderedFunc = (obj: object, keyName: string) => void;
+
+export const runInTransaction: RunInTransactionFunc;
```

Comment:
I don't grok why we need this (along with the other lines above), I would have thought you could replace the preexisting `export function runInTransaction` line with `export { runInTransaction } from './lib/transaction';` and all of these types in this `index.d.ts` wouldn't need to be redefined. What am I missing?

Reply:
@rwjblue You're right. I was over-thinking it. Fixed in latest commit.